zio-http/src/main/scala/zio/http/middleware/Metrics.scala

```diff
@@ -0,0 +1,76 @@
+package zio.http.middleware
+
+import zio.http.{Http, Middleware, Request, Response}
+import zio._
+import zio.metrics.{Metric, MetricKeyType, MetricLabel}
+
+private[zio] trait Metrics {
+  def metrics(
+    pathLabelMapper: PartialFunction[Request, String] = Map.empty,
```

Comment:
Can be used to override mappings (i.e map certain paths like /users/1 to /users/:id) to avoid having Prometheus label count explode (and thus crippling its performance).